atoma-auth/src/auth.rs

```diff
@@ -522,11 +538,13 @@ impl Auth {
             .map(|c| {
                 BASE64_URL_CHARSET
                     .find(c)
-                    .map(|index| index as u8)
-                    .map(|index| (0..6).rev().map(move |i| index >> i & 1))
-                    .ok_or_else(|| {
-                        AuthError::AnyhowError(anyhow!("base64_to_bitarry invalid input"))
+                    .map(|index| {
+                        u8::try_from(index).map_err(|e| {
+                            AuthError::AnyhowError(anyhow!("Failed to convert index: {e}"))
```

Comment:
You could add a new enum field for the `TryFrom` type `Error`and we would not need to call `map_err`multiple times here

Reply:
That's true, I think that's a cleaner approach 6104ade